bridge/engine/ActiveMediaList.cpp

```diff
@@ -481,41 +542,39 @@ void ActiveMediaList::process(const uint64_t timestamp,
     }
     _activeTalkerSnapshot.write(activeTalkersSnapshot);
 
-    if (timestamp - _lastChangeTimestamp + 10 * utils::Time::ms * (requiredConsecutiveWins - 1) <
-        maxSwitchDominantSpeakerEvery)
+    if (outAudioMapChanged && logger::_logLevel >= logger::Level::DBG)
     {
-        return;
+        logAudioList();
     }
 
-    // nominate dominant speaker (for video)
-    if (dominantSpeaker.participant == _prevWinningDominantSpeaker)
+    if (nominatedSpeaker.score == 0.0)
     {
-        _consecutiveDominantSpeakerWins++;
+        return;
     }
-    else
+
+    if (nominatedSpeaker.participant != _nominatedSpeaker)
     {
-        _consecutiveDominantSpeakerWins = 1;
-        _prevWinningDominantSpeaker = dominantSpeaker.participant;
+        _nominationTimestamp = timestamp;
+        _nominatedSpeaker = nominatedSpeaker.participant;
     }
 
-    _lastRunTimestamp = timestamp;
-    if (dominantSpeaker.participant != _dominantSpeakerId &&
-        ((!_dominantSpeakerId || currentDominantSpeakerScore < 0.01) ||
-            (_consecutiveDominantSpeakerWins >= requiredConsecutiveWins &&
-                currentDominantSpeakerScore < 0.75 * dominantSpeaker.score &&
-                timestamp - _lastChangeTimestamp >= maxSwitchDominantSpeakerEvery)))
+    // switch at most every 2s
```

Comment:
simplified condition. Dominant speaker gets grace period of 2s.
Bargein can happen either if dominant stops speaking or you try to bargein for 2s.